internal/adminapi/kong.go

```diff
@@ -43,7 +43,12 @@ func (e KongGatewayUnsupportedVersionError) Error() string {
 
 // NewKongAPIClient returns a Kong API client for a given root API URL.
 // It ensures that proper User-Agent is set. Do not use kong.NewClient directly.
-func NewKongAPIClient(adminURL string, httpClient *http.Client) (*kong.Client, error) {
+func NewKongAPIClient(adminURL string, kongAdminAPIConfig ClientOpts, kongAdminToken string) (*kong.Client, error) {
```

Comment:
Food for thought: making the `adminURL` a [`url.URL`](https://pkg.go.dev/net/url#URL) (that could be done in a separate PR).

This will avoid the conversion in call sites that use `url.URL` and also make impossible to mix the admin token (last param) with this one.

Reply:
I'll to in separate one